src/Vivait/LicensingClientBundle/Controller/TokenController.php

```diff
@@ -24,11 +24,20 @@ public function tokenAction(Request $request)
         $licensingApi = $this->get('vivait_licensing_client.licensing.api');
 
         try {
-            $tokenData = $licensingApi->getToken(
-                $request->query->get('client_id', null),
-                $request->query->get('client_secret', null),
-                $request->query->get('grant_type', 'client_credentials')
-            );
+            if($request->getMethod() == 'POST') {
+                $tokenData = $licensingApi->getToken(
+                    $request->request->get('client_id', null),
+                    $request->request->get('client_secret', null),
+                    $request->request->get('grant_type', 'client_credentials')
+                );
+            } else {
+                $tokenData = $licensingApi->getToken(
+                    $request->query->get('client_id', null),
+                    $request->query->get('client_secret', null),
+                    $request->query->get('grant_type', 'client_credentials')
+                );
+     
+            }
             $clientData = $licensingApi->getClient($tokenData['access_token']);
```

Comment:
Sorry to nit-pick but the duplication could be avoided by assigning the variable, e.g.:

```
if($request->getMethod() == 'POST') {
    $parameters =  $request->request;
}
else {
    $parameters =  $request->query;
}

$tokenData = $licensingApi->getToken(
                    $parameters->get('client_id', null),
                    $parameters->query->get('client_secret', null),
                    $parameters->query->get('grant_type', 'client_credentials')
                );
```

Then it avoids the off-chance that someone changes one but not the other. What do you think?